modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/AggregateProcess.java

```diff
@@ -454,5 +475,103 @@ public List<Object[]> getGroupByResult() {
         public EnumMap<AggregationFunction, Number> getResults() {
             return results;
         }
+
+        /**
+         * Converts the results to a {@link SimpleFeatureCollection} where each feature represents a group and contains
+         * the aggregated values as attributes. If the result is not from a grouping, no feature collection will be
+         * returned (null).
+         *
+         * @return a {@link SimpleFeatureCollection} representing the aggregation results
+         */
+        public SimpleFeatureCollection toFeatureCollection() {
+            if (schema == null || groupByAttributes == null || groupByAttributes.isEmpty()) return null;
+
+            SimpleFeatureType outputSchema = buildOutputSchema();
+            SimpleFeatureBuilder fb = new SimpleFeatureBuilder(outputSchema);
+
+            ListFeatureCollection collection = new ListFeatureCollection(outputSchema);
+            for (Object[] objects : groupByResult) {
+                fb.addAll(objects);
+                SimpleFeature sf = fb.buildFeature(null);
+                collection.add(sf);
+            }
+
+            return collection;
+        }
+
+        private SimpleFeatureType buildOutputSchema() {
+            // add types for the grouping attributes
+            SimpleFeatureTypeBuilder tb = new SimpleFeatureTypeBuilder();
+            for (String att : groupByAttributes) {
+                AttributeDescriptor ad = schema.getDescriptor(att);
+                checkAttributeDescriptor(att, ad);
+                tb.add(ad);
+            }
+
+            // check the aggregate attribute
+            AttributeDescriptor ad = schema.getDescriptor(aggregateAttribute);
+            checkAttributeDescriptor(aggregateAttribute, ad);
+            Class<?> type = ad.getType().getClass();
+
+            // add types for the aggregation functions
+            for (AggregationFunction function : functions) {
+                switch (function) {
+                    case Count:
+                        tb.add(function.name(), Long.class);
+                        break;
+                    case Average:
+                    case StdDev:
+                    case Sum:
+                    case SumArea:
+                        tb.add(function.name(), Double.class);
+                        break;
+                    case Max:
+                    case Median:
```

Comment:
Is the median not applying any "math" on the values (just selecting either one or the other) so that it relies on the type?
(Therefore not using Double as per other aggregators)